NetworkingServiceKit/Classes/ServiceLocator.swift

```diff
@@ -37,20 +37,22 @@ open class ServiceLocator: NSObject {
     private var token: APIToken?
 
     /// Inits default configuration and network manager
-    private override init() {
+    private init(delegate: ServiceLocatorDelegate? = nil) {
+        self.delegate = delegate
         self.currentServices = [String: Service]()
         self.loadedServiceTypes = [Service.Type]()
     }
 
     /// Resets the ServiceLocator singleton instance
     open class func reset() {
-        ServiceLocator.shared = ServiceLocator()
+        ServiceLocator.shared = ServiceLocator(delegate: ServiceLocator.shared.delegate)
     }
 
     /// Reloads token, networkManager and configuration with existing hooked services
     open class func reloadExistingServices() {
+        reset()
```

Comment:
You don't want this behavior that's why they are separate funcs. You could be hard linking to a SL instance expecting only the services to be deallocated but no the SL itself